data/ha/hawk_test.py

```diff
@@ -757,52 +45,45 @@ def ssh_verify_primitive_removed(ssh):
 myclone = args.prefix.lower() + 'cool_clone'
 mygroup = args.prefix.lower() + 'cool_group'
 
-# Variables for texts that change between chrome and firefox drivers
-troubleshoot = link_by_browser(browser, 'troubleshooting')
-monitoring = link_by_browser(browser, 'monitoring')
-
 # Tests to perform
-browser.get(mainlink)
-hawk_log_in(browser)
-set_stonith_maintenance(browser, timeout_scale)
+browser.test('test_set_stonith_maintenance', results)
 if args.secret:
-    ssh_verify_stonith(ssh)
-disable_stonith_maintenance(browser, timeout_scale)
-view_details_first_node(browser, timeout_scale)
-clear_state_first_node(browser, timeout_scale)
-set_first_node_maintenance(browser, timeout_scale)
+    ret = ssh.verify_stonith_in_maintenance(results)
+    if ret != 0:
+        browser.set_retval(ret)
+browser.test('test_disable_stonith_maintenance', results)
+browser.test('test_view_details_first_node', results)
+browser.test('test_clear_state_first_node', results)
+browser.test('test_set_first_node_maintenance', results)
 if args.secret:
-    ssh_verify_node_maintenance(ssh)
-disable_maintenance_first_node(browser, timeout_scale)
-add_new_cluster(browser, mycluster, timeout_scale)
-remove_cluster(browser, mycluster, timeout_scale)
-test_click_on(browser, 'Dashboard', timeout_scale)
-click_if_major_version(browser, args.test_version.lower(), "15", troubleshoot, timeout_scale)
-# Only click on History and on Generate Report in Firefox if testing 15+
-if args.browser in ['chrome', 'chromium'] or Version(args.test_version.lower()) >= Version('15'):
-    test_click_on(browser, 'History', timeout_scale, 'test_click_on_history')
-    generate_report(browser, timeout_scale)
-test_click_on(browser, 'Command Log', timeout_scale, 'test_click_on_command_log')
-click_if_major_version(browser, args.test_version.lower(), "15", monitoring, timeout_scale)
-test_click_on(browser, 'Status', timeout_scale, 'test_click_on_status')
-add_primitive(browser, myprimitive, timeout_scale, args.test_version.lower())
+    ret = ssh.verify_node_maintenance(results)
+    if ret != 0:
+        browser.set_retval(ret)
```

Comment:
this would just override retval which you setting in line 53 . is this expected ?
can you generally explain purpose of this retval field ? it is not clear to me why not all tests using it

Reply:
Yes, if any of the tests or SSH verification fail, the return value for the script is overridden. Otherwise the previous value (whether it's the 0 default or a non-zero value representing a failed test) remains. I think now that these methods are basically returning 0 on success and 1 on failure, the code could be clearer with True/False.

Reply:
you saying that if **any** of the tests fail but I see only some of them using set_retval so what the logic behind ? all additionally to retval you have results which also contains information about tests results so what relation between results and retval and why you need both ?

Reply:
Definitely something from a previous version. I will remove values different than 0 or 1 for retval throughout the code, as I don't think that is needed anymore.